tortoise/ninja_tortoise.go

```diff
@@ -89,18 +102,18 @@ type ninjaTortoise struct {
 	TCorrect     map[types.BlockID]map[types.BlockID]vec           //correction vectors
 	TExplicit    map[types.BlockID]map[types.LayerID]votingPattern //explict votes from block to layer pattern
 
-	TSupport           map[votingPattern]int             //for pattern p the number of blocks that support p
-	TComplete          map[votingPattern]struct{}        //complete voting Patterns
-	TEffectiveToBlocks map[votingPattern][]types.BlockID //inverse blocks effective pattern
+	TSupport           map[votingPattern]int                 //for pattern p the number of blocks that support p
+	TComplete          map[votingPattern]struct{}            //complete voting Patterns
+	TEffectiveToBlocks map[votingPattern][]BlockIDLayerTuple //inverse blocks effective pattern
```

Comment:
not suggesting we should do it now be technically it could also be 
`map[votingPattern]map[types.LayerID][]types.BlockID` right ?

Reply:
could but its simpler this way